webapp/src/ts/services/xml-forms.service.ts

```diff
@@ -236,8 +236,16 @@ export class XmlFormsService {
 
     return this
       .filterContactTypes(form.context, options.doc)
-      .then(valid => valid && this.checkFormPermissions(form))
-      .then(valid => valid && this.checkFormExpression(form, options.doc, user, options.contactSummary));
+      .then(valid => valid && this.canAccessForm(form, user, options));
+  }
+
+  async canAccessForm(form, userContact?, options?) {
+    if (!await this.checkFormPermissions(form)) {
+      return false;
+    }
+
+    const user = userContact || await this.userContactService.get();
```

Comment:
Ohh, I think I see. If it is an admin user without a contact, this might be null.  Still, in that case we are never going to get a value back from `userContactService.get()`, so I still don't think we need this check.